sensor/common/delegatedregistry/delegated_registry_handler.go

```diff
@@ -0,0 +1,160 @@
+package delegatedregistry
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/images/utils"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/registry"
+	"github.com/stackrox/rox/sensor/common/scan"
+	"google.golang.org/grpc"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	scanTimeout         = 6 * time.Minute
+	statusUpdateTimeout = 10 * time.Second
+)
+
+// Handler is responsible for processing delegated registry related requests
+// from Central.
+type Handler interface {
+	common.SensorComponent
+}
+
+type delegatedRegistryImpl struct {
+	registryStore *registry.Store
+	stopSig       concurrency.Signal
+	localScan     *scan.LocalScan
+	imageSvc      v1.ImageServiceClient
+}
+
+// NewHandler returns a new instance of Handler.
+func NewHandler(registryStore *registry.Store, localScan *scan.LocalScan) Handler {
+	return &delegatedRegistryImpl{
+		registryStore: registryStore,
+		stopSig:       concurrency.NewSignal(),
+		localScan:     localScan,
+	}
+}
+
+func (d *delegatedRegistryImpl) Capabilities() []centralsensor.SensorCapability {
+	if !env.LocalImageScanningEnabled.BooleanSetting() {
+		// Do not advertise the capability if local scanning is disabled.
+		return nil
+	}
+
+	return []centralsensor.SensorCapability{centralsensor.DelegatedRegistryCap}
+}
+
+func (d *delegatedRegistryImpl) Notify(_ common.SensorComponentEvent) {}
+
+func (d *delegatedRegistryImpl) ProcessMessage(msg *central.MsgToSensor) error {
+	if !env.LocalImageScanningEnabled.BooleanSetting() {
+		// Ignore all messages if local scanning is disabled.
+		return nil
+	}
+
+	switch {
+	case msg.GetDelegatedRegistryConfig() != nil:
+		return d.processUpdatedDelegatedRegistryConfig(msg.GetDelegatedRegistryConfig())
+	case msg.GetScanImage() != nil:
+		return d.processScanImage(msg.GetScanImage())
+	}
+
+	return nil
+}
+
+func (d *delegatedRegistryImpl) ResponsesC() <-chan *central.MsgFromSensor {
+	return nil
+}
+
+func (d *delegatedRegistryImpl) Start() error {
+	return nil
+}
+
+func (d *delegatedRegistryImpl) Stop(_ error) {
+	d.stopSig.Signal()
+}
+
+func (d *delegatedRegistryImpl) processUpdatedDelegatedRegistryConfig(config *central.DelegatedRegistryConfig) error {
+	select {
+	case <-d.stopSig.Done():
+		return errors.New("could not process updated delegated registry config, stop requested")
+	default:
+		d.registryStore.SetDelegatedRegistryConfig(config)
+		log.Infof("Stored updated delegated registry config: %q", config)
```

Comment:
Same notion here: Info or debug going forward?

Reply:
`Info` was intentional. I started with all the logs being `Debug` throughout the PR and changed the few to `Info` based on anticipating user issues.  In many cases the log dumps shared are at Info level, felt this would help troubleshoot in addition to updating the config is likely going to be a rare occurrence I didn't think it would be an issue to log at `Info`.  Happy to change if you feel differently.